core/trino-main/src/main/java/io/trino/execution/querystats/QueryStatsCollector.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.querystats;
+
+import io.trino.sql.planner.iterative.Rule;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+
+public interface QueryStatsCollector
```

Comment:
Interface looks unnecessary to me.
You can add the below method to the implementation class and use that everywhere in tests.
```
public static QueryStatsCollector createTestingOptimizerStatsCollector()
{
     return new QueryStatsCollector(Integer.MAX_VALUE);
}
```